tests/src/string/src/main.sw

```diff
@@ -442,170 +428,161 @@ impl StringTest for Contract {
-        // assert(string.len() == 0);
-        // assert(string.is_empty());
-        // string.push(NUMBER1);
-        // assert(string.nth(0).unwrap() == NUMBER1);
-        // assert(string.len() == 1);
-
-        // string.push(NUMBER1);
-        // assert(string.nth(1).unwrap() == NUMBER1);
-        // assert(string.len() == 2);
-
-        // string.push(NUMBER0);
-        // assert(string.nth(2).unwrap() == NUMBER0);
-        // assert(string.len() == 3);
+        string.push(NUMBER0);
+        assert(string.nth(0).unwrap() == NUMBER0);
+        assert(string.len() == 1);
+
+        string.push(NUMBER1);
+        assert(string.nth(1).unwrap() == NUMBER1);
+        assert(string.len() == 2);
+
+        string.push(NUMBER2);
+        assert(string.nth(2).unwrap() == NUMBER2);
+        assert(string.len() == 3);
+
+        string.push(NUMBER3);
+        assert(string.nth(3).unwrap() == NUMBER3);
+        assert(string.len() == 4);
+
+        string.push(NUMBER4);
+        assert(string.nth(4).unwrap() == NUMBER4);
+        assert(string.len() == 5);
+
+        string.push(NUMBER5);
+        assert(string.nth(5).unwrap() == NUMBER5);
+        assert(string.len() == 6);
+
+        string.push(NUMBER6);
+        assert(string.nth(6).unwrap() == NUMBER6);
+        assert(string.len() == 7);
+
+        string.push(NUMBER7);
+        assert(string.nth(7).unwrap() == NUMBER7);
+        assert(string.len() == 8);
+
+        string.push(NUMBER8);
+        assert(string.nth(8).unwrap() == NUMBER8);
+        assert(string.len() == 9);
+
+        string.push(NUMBER1);
+        assert(string.nth(9).unwrap() == NUMBER1);
+        assert(string.len() == 10);
+
+        string.clear();
+        assert(string.len() == 0);
+        assert(string.is_empty());
+        string.push(NUMBER1);
+        assert(string.nth(0).unwrap() == NUMBER1);
+        assert(string.len() == 1);
+
+        string.push(NUMBER1);
+        assert(string.nth(1).unwrap() == NUMBER1);
+        assert(string.len() == 2);
+
+        string.push(NUMBER0);
+        assert(string.nth(2).unwrap() == NUMBER0);
+        assert(string.len() == 3);
     }
 
     fn test_set() {
+        let mut string = String::new();
 
-        // Uncomment when https://github.com/FuelLabs/sway/issues/4408 is resolved
-        // let mut string = String::new();
-
-        // string.push(NUMBER0);
-        // string.push(NUMBER1);
-        // string.push(NUMBER2);
+        string.push(NUMBER0);
+        string.push(NUMBER1);
+        string.push(NUMBER2);
 
-        // assert(string.nth(0).unwrap() == NUMBER0);
-        // assert(string.nth(1).unwrap() == NUMBER1);
-        // assert(string.nth(2).unwrap() == NUMBER2);
+        assert(string.nth(0).unwrap() == NUMBER0);
+        assert(string.nth(1).unwrap() == NUMBER1);
+        assert(string.nth(2).unwrap() == NUMBER2);
 
-        // string.set(0, NUMBER3);
-        // string.set(1, NUMBER4);
-        // string.set(2, NUMBER5);
+        string.set(0, NUMBER3);
```

Comment:
Since we are checking the setting we should assert after each call otherwise it could be that after the call to `set` 3 times with any values could result in a false positive pass of this test because of the underlying implementation